### PR TITLE
Preventing cyclical calling of building route in case of some errors.

### PR DESCRIPTION
**Pull request by bykoianko**
@dobriy-eeh @ygorshenin PTAL

### CODE REVIEW

**Comment by dobriy-eeh** [comment]
LGTM

File: routing/car_router.cpp [hunk 1]
```diff
@@ -282,7 +282,16 @@ bool CarRouter::FindRouteMSMT(TFeatureGraphNodeVec const & sources,
       {
       case NoError: return true;
       case Cancelled: return false;
-      default: continue;
+      case NoCurrentPosition: return false;
```
Comment (ygorshenin) [inline routing/car_router.cpp]: But sources are specified as a parameter, right?

Comment (bykoianko) [inline routing/car_router.cpp]: There's only one place where something is called:
```c++
void Framework::BuildRoute(m2::PointD const & finish, uint32_t timeoutSec)
{
  ASSERT_THREAD_CHECKER(m_threadChecker, ("BuildRoute"));
  ASSERT(m_drapeEngine != nullptr, ());

  m2::PointD start;
  if (!m_drapeEngine->GetMyPosition(start))
  {
    CallRouteBuilded(IRouter::NoCurrentPosition, storage::TCountriesVec());
    return;
  }
  BuildRoute(start, finish, false /* isP2P */, timeoutSec);
}
```
So, I guess it's possible to put an ASSERT here.

**Comment by bykoianko** [comment]
@ygorshenin PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**